### PR TITLE
[Go] Improve tests: fix test issues, add cases and locate assets 

### PR DESCRIPTION
**Pull request by wamuir**
PR makes improvements to tensorflow/go tests:
- Fixes several missing checks on returned errors
- Adds test cases for LoadSavedModel (half plus two) and improves consistency of variable naming with funcs and documentation for api
- Locates a minimal set of required test assets inside the module; this addresses issues caused by making relative references elsewhere in the repository and outside of the Go module
- Makes a few grammatical corrections

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
This keeps failing internal tests. Let me see if I can quickly generate a fix

**Comment by mihaimaruseac** [comment]
Fix generated, now waiting for approval internally.

**Pull request merged**
